regulation-worker/internal/delete/batch/batch.go

```diff
@@ -550,13 +435,63 @@ func (bm *BatchManager) Delete(ctx context.Context, job model.Job, destConfig ma
 		}
 		err = g.Wait()
 		if err != nil {
-			pkgLogger.Errorf("job failed with error: %v", err)
+			pkgLogger.Errorf("user identity deletion job failed with error: %v", err)
 			return model.JobStatusFailed
 		}
 	}
+
 	return model.JobStatusComplete
 }
 
+// getFileHandler extracts the filehandler based on the suffix of the file for which
+// we need to perform the identity removal.
+func getFileHandler(key string, handlers map[string]filehandler.LocalFileHandler) filehandler.LocalFileHandler {
+	// handlers map are over the file suffix like .json.gz, .parquet
+	// based on the file suffix, allow for the fetch of corresponding handler.
+	for k, v := range handlers {
+		if strings.HasSuffix(key, k) {
+			return v
+		}
+	}
+
+	return nil
+}
+
+// handleIdentityRemoval is a convenience wrapper over the filehandler
+// performing the operations over the file to remove the user identity.
+func handleIdentityRemoval(
+	ctx context.Context,
+	handler filehandler.LocalFileHandler,
+	attributes []model.User,
+	sourceFile, targetFile string,
+) error {
+	pkgLogger.Debugf("Handling identity removal for source: %s, destination: %s", sourceFile, targetFile)
+
+	if err := handler.Read(ctx, sourceFile); err != nil {
+		return fmt.Errorf("unable to parse contents of local file: %s, err: %w", sourceFile, err)
+	}
+
+	if err := handler.RemoveIdentity(ctx, attributes); err != nil {
+		return fmt.Errorf("unable to handle identity removal for attributes: %v, err: %w", nil, err)
+	}
+
+	if err := handler.Write(ctx, targetFile); err != nil {
+		return fmt.Errorf("unable to write to local file: %s, err: %w", targetFile, err)
```

Comment:
[minor] idiomatic errors
```suggestion
		return fmt.Errorf("parse contents of local file: %s, err: %w", sourceFile, err)
	}

	if err := handler.RemoveIdentity(ctx, attributes); err != nil {
		return fmt.Errorf("identity removal for attributes: %v, err: %w", nil, err)
	}

	if err := handler.Write(ctx, targetFile); err != nil {
		return fmt.Errorf("write to local file: %s, err: %w", targetFile, err)
```